xxhash.h

```diff
@@ -2041,12 +2041,20 @@ static int XXH_isLittleEndian(void)
 #endif
 
 
-#if defined(__STDC_VERSION__) && (__STDC_VERSION__ > 201710L)
-/* C23 and future versions have standard "unreachable()" */
-#  include <stddef.h>
-#  define XXH_UNREACHABLE() unreachable()
 
-#elif defined(__cplusplus) && (__cplusplus > 202002L)
+/*
+ * C23 and future versions have standard "unreachable()".
+ * Once it has been implemented reliably we can add it as an
+ * additional case:
+ *
+ * ```
+ * #if defined(__STDC_VERSION__) && (__STDC_VERSION__ > 201710L)
```

Comment:
I presume you meant `__STDC_VERSION__ > 202310L` in this example

Reply:
Truthfully not sure. The code was based on what @t-mat suggested [here](https://github.com/Cyan4973/xxHash/pull/754#issuecomment-1312366653).

I can drop the comment entirely if its liable to make confusion.

Reply:
Since C23 didn't set actual `__STDC_VERSION__` yet (right?), I think we can express C23 as "beyond/greater than C17 (`> 201710L`)" for now.
It must be fixed after the comittee and compilers set official `__STDC_VERSION__` though.
So, adding `TODO` or `note` comment would be nice to reminder.